nixos/modules/hardware/coral.nix

```diff
@@ -0,0 +1,44 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkMerge
+    ;
+
+  cfg = config.hardware.coral;
+in
+
+{
+  options.hardware.coral = {
+    usb.enable = mkEnableOption "Coral USB support";
+    pcie.enable = mkEnableOption "Coral PCIe support";
+  };
+
+  config = mkMerge [
+    (mkIf (cfg.usb.enable || cfg.pcie.enable) {
+      users.groups.coral = { };
+    })
+    (mkIf cfg.usb.enable {
+      services.udev.packages = with pkgs; [ libedgeptu ];
+    })
+    (mkIf cfg.pcie.enable {
+      boot = {
+        kernelPackages = with config.boot.kernelPackages; [ gasket ];
+        kernelModules = [
```

Comment:
I think preloading the kernel modules isn't necessary.  Or at least I don't need it for my coral pcie tpu.